app/src/main/java/com/duckduckgo/app/browser/BrowserChromeClient.kt

```diff
@@ -116,4 +116,9 @@ class BrowserChromeClient @Inject constructor(private val uncaughtExceptionRepos
         }
         return false
     }
+
+    override fun onCloseWindow(window: WebView?) {
+        webViewClientListener?.closeCurrentTabIfOpenedByPage()
```

Comment:
What's your motivation for making this conditional rather than just closing the tab entirely? The api for this at https://developer.android.com/reference/android/webkit/WebChromeClient#onCloseWindow(android.webkit.WebView) states:

> Notify the host application to close the given WebView and remove it from the view system if necessary. At this point, WebCore has stopped any loading in this window and has removed any cross-scripting ability in javascript.

If we can just close it a lot of complexity (e.g state storage, db migrations) and will disappear.

Reply:
Ugh, silly me. I stupidly assumed it behaved like WKWebView.  I'll roll it back and put a simpler implementation.

Reply:
Actually, I tested the simple solution and window.close gets triggered in the case where you open a new tab, go to close.html directly and click the button.  Chrome doesn't appear to behave like that so we need to make a call what is the desired the behaviour. 

Talk to you about it over Zoom tomorrow.